Lib/unittest/case.py

```diff
@@ -1321,15 +1321,77 @@ def assertIsInstance(self, obj, cls, msg=None):
         """Same as self.assertTrue(isinstance(obj, cls)), with a nicer
         default message."""
         if not isinstance(obj, cls):
-            standardMsg = '%s is not an instance of %r' % (safe_repr(obj), cls)
+            if isinstance(cls, tuple):
+                standardMsg = '%s is not an instance of any of %r' % (safe_repr(obj), cls)
```

Comment:
You may use f-strings :-)

Reply:
Actually, `'%s is not an instance of any of %r' % (safe_repr(obj), cls)` and `f'{safe_repr(obj)!s} is not an instance of any of {cls!r}'` produce the same bytecode. So the difference is only in readability, which is at large part subjective. I was not sure that inlining expressions in f-strings would make the code more readable, but if you think so...